### PR TITLE
SDIT-642 Everything now reactive or suspend

### PR DESCRIPTION
**Pull request by andymarke**
* no blocking calls at webClient
* wrap tests in `runTest`
* change assertThrownBy to assertThrows that supports suspend

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/health/HealthCheck.kt [hunk 1]
```diff
@@ -2,23 +2,26 @@ package uk.gov.justice.digital.hmpps.prisonertonomisupdate.health
...
 
-  override fun health(): Health? {
+  override fun health(): Mono<Health> {
     return webClient.get()
       .uri("/health/ping")
       .retrieve()
       .toEntity(String::class.java)
       .flatMap { Mono.just(Health.up().withDetail("HttpStatus", it?.statusCode).build()) }
-      .onErrorResume(WebClientResponseException::class.java) { Mono.just(Health.down(it).withDetail("body", it.responseBodyAsString).withDetail("HttpStatus", it.statusCode).build()) }
+      .onErrorResume(WebClientResponseException::class.java) {
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/health/HealthCheck.kt:19]: just formatting

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/services/NomisApiService.kt [hunk 2]
```diff
@@ -203,10 +197,6 @@ data class UpdateVisitDto(
   val openClosedStatus: String,
 )
 
-data class CreateVisitResponseDto(
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/services/NomisApiService.kt:206]: not used

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesApiServiceTest.kt [hunk 3]
```diff
@@ -1,11 +1,15 @@
+@file:OptIn(ExperimentalCoroutinesApi::class)
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesApiServiceTest.kt:1]: not sure why you need these but you do, e.g not sure why a test helper is "experimental"

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesServiceTest.kt [hunk 4]
```diff
@@ -513,19 +517,21 @@ internal class ActivitiesServiceTest {
...
-        activitiesService.createAllocation(
-          AllocationDomainEvent(
-            eventType = "dummy",
-            version = "1.0",
-            description = "description",
-            occurredAt = LocalDateTime.now(),
-            additionalInformation = AllocationAdditionalInformation(
-              allocationId = ALLOCATION_ID,
+      assertThat(
+        assertThrows<RuntimeException> {
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesServiceTest.kt:521]: I noticed you replaced this with assertThatThrownBy elsewhere. Just wondering why?

Comment (mikehalmamoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesServiceTest.kt:521]: Nope I'm talking rubiish! Please ignore

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**